tools/mo/requirements.txt

```diff
@@ -1,6 +1,8 @@
-numpy>=1.16.6
-networkx~=2.5; python_version <= "3.6"
-networkx<=2.8.8; python_version > "3.6"
-defusedxml>=0.7.1
-importlib-metadata; python_version < "3.8" and sys_platform == 'win32'
-openvino-telemetry>=2022.1.0
\ No newline at end of file
+-c ../../src/bindings/python/constraints.txt
+numpy>=1.16.6,<1.25.0
```

Comment:
disagree with this change:
1. please remove upper-bound
2. why do we need the constrain files here? This requirements file is going to the release package and now we have relative path that can be invalidated in the release package

Reply:
> 1. please remove upper-bound for NumPy

All of the requirement files in `tools/mo` have upper-bound requirement, except for `requirements.txt`. Approach proposed in this PR allows us to specify one numpy version requirement for MO, but if you wish I can move it back to the previous state (where no upper-bound is present only in `requirements.txt`.

> 2. why do we need the constrain files here? This requirements file is going to the release package and now we have relative path that can be invalidated in the release package

That's a good point, I'm currently testing the `openvino_dev` wheel.

Reply:
Please remove upper-bound in `requirements.txt` because it corresponds to the default installation. We agreed to have the default installation preserving user environment without overriding NumPy version

Reply:
I understand now. Upper bound has been removed.